soroban-react-dapp/contracts/greeting/src/lib.rs

```diff
@@ -1,24 +1,96 @@
 #![no_std]
-use soroban_sdk::{contract, contractimpl, Env, Symbol, symbol_short, String};
-
-const TITLE: Symbol = symbol_short!("TITLE");
-
+use soroban_sdk::{contract, contracterror, contractimpl, contracttype, Address, Env, String, Vec};
 
 #[contract]
 pub struct TitleContract;
 
+#[contracterror]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(u32)]
+pub enum Error {
+    Unauthorized = 1,
+    AlreadyInitialized = 2,
+}
+
 #[contractimpl]
 impl TitleContract {
+    // initialize the contract and set the admin
+    pub fn init(env: Env, admin: Address) -> Result<(), Error> {
+        let storage = env.storage().instance();
+        admin.require_auth();
+        if storage.has(&Assets::Admin) {
+            return Err(Error::AlreadyInitialized);
+        }
+        storage.set(&Assets::Admin, &admin);
+        Ok(())
+    }
 
-    pub fn set_title(env: Env, title: String) {
-                env.storage().instance().set(&TITLE, &title)
+    // set the title only available editors
+    pub fn set_title(env: Env, user: Address, title: String) -> Result<(), Error> {
+        user.require_auth();
+        let storage = env.storage().instance();
+        let admin: Address = storage.get(&Assets::Admin).unwrap();
```

Comment:
This will fail without Error message if admin has not been set yet, hence, if the contrract has not been initialized yet

Reply:
got it. I'll fix it with error handling

Reply:
let admin: Address = storage.get(&Assets::Admin).unwrap_or_else(|| {
            // You can log or handle the error here
            // In this case, we'll return an error
            return Err(Error::Unauthorized);
        })?;